### PR TITLE
Add optional name property to testgroup for better test failure output

### PR DESCRIPTION
**Pull request by mishamo**
Addresses #8439

### CODE REVIEW

**Comment by roidelapluie** [comment]
Thank you for this. This is becoming a bit unreadable, maybe we could try an alternative approach to build the error message, like a string.Builder?

File: cmd/promtool/unittest.go [hunk 1]
```diff
@@ -299,16 +300,29 @@ func (tg *testGroup) test(evalInterval time.Duration, groupOrderMap map[string]i
 					})
 				}
 
+				var sb strings.Builder
 				if gotAlerts.Len() != expAlerts.Len() {
-					errs = append(errs, errors.Errorf("    alertname:%s, time:%s, \n        exp:%#v, \n        got:%#v",
-						testcase.Alertname, testcase.EvalTime.String(), expAlerts.String(), gotAlerts.String()))
+					if tg.TestGroupName != "" {
+						fmt.Fprintf(&sb, "    name: %s,\n", tg.TestGroupName)
```
Comment (roidelapluie) [inline cmd/promtool/unittest.go:306]: can you use sb.Write()

**Comment by mishamo** [comment]
I'm not sure what's happening with the `ci/circleci: test_windows` check. I don't believe it's related to this PR.

**Review by roidelapluie — APPROVED** [review]
Thanks! I will leave @dgl and @simonpasquier the opportunity to add remarks, otherwise I will merge this tomorrow.

**Comment by mishamo** [comment]
Thanks. Should something also be added to the docs as well? Can I leave that part to you guys?

**Comment by roidelapluie** [comment]
The change should be done here: docs/configuration/unit_testing_rules.md

I think mentioning the field in test_group should be enough,something like

```
# Name of the test group.
[ name: <string> ]`
```

File: docs/configuration/unit_testing_rules.md [hunk 2]
```diff
@@ -44,6 +44,9 @@ interval: <duration>
 input_series:
   [ - <series> ]
 
+#Name of the test group
```
Comment (roidelapluie) [inline docs/configuration/unit_testing_rules.md:47]:
```suggestion
# Name of the test group.
```

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**